### PR TITLE
Implement PCR logging in FMC

### PR DESCRIPTION
**Pull request by bluegate010**
Based on https://github.com/chipsalliance/caliptra-sw/pull/722

Also changes FMC to extend PCRs only on cold or update reset, as per https://github.com/chipsalliance/caliptra-sw/issues/776.

### CODE REVIEW

File: fmc/README.md [hunk 1]
```diff
@@ -136,6 +136,7 @@ fields may not be changed or removed). Table revisions with different Major Vers
 | ldevid_tbs_size       | 2            | ROM        | Local Device ID TBS Size.                                                                                |
 | fmcalias_tbs_size     | 2            | ROM        | FMC Alias TBS Size.                                                                                      |
 | pcr_log_addr          | 4            | ROM        | PCR Log Address.                                                                                         |
+| pcr_log_index         | 4            | ROM        | Last empty PCR log entry slot index.                                                                     |
```
Comment (FerralCoder) [inline fmc/README.md:139]: also needs a more detailed description in the section just below this

**Review by FerralCoder — APPROVED** [review]

File: fmc/README.md (hunk 1 above)
Comment (bluegate010) [reply fmc/README.md:139]: Done. I noticed there are quite a few fields missing, have added TODOs for them.

Comment (FerralCoder) [reply fmc/README.md:139]: Thanks, I've been trying to stay on top of this but I guess I didn't catch them all. Been meaning to go back and make sure the descriptions were all there.

**Review by FerralCoder — DISMISSED** [review]

**Review by FerralCoder — APPROVED** [review]

File: fmc/src/flow/pcr.rs [hunk 2]
```diff
@@ -23,49 +23,87 @@ Note:
...
+    pcr_ids: u32,
+    data: &[u8],
+) -> CaliptraResult<()> {
+    let Some(dst) = pcr_log.get_mut(pcr_bank.log_index) else {
+        return Err(CaliptraError::FMC_GLOBAL_PCR_LOG_EXHAUSTED);
+    };
+
+    // Create a PCR log entry
+    cprintln!("pcr_entry_id: {:?}", pcr_entry_id as u16);
+    cprintln!("FMC log_index: {:?}", pcr_bank.log_index as u16);
```
Comment (jhand2) [inline fmc/src/flow/pcr.rs:93]: Can these be removed? The information is in the PCR log anyway.

File: fmc/src/flow/pcr.rs [hunk 3]
```diff
@@ -23,49 +23,87 @@ Note:
...
-///
-/// # Arguments
-///
-/// * `env` - FMC Environment
-pub fn extend_journey_pcr(env: &mut FmcEnv, hand_off: &HandOff) -> CaliptraResult<()> {
-    extend_pcr_common(env, hand_off, RT_FW_JOURNEY_PCR)
-}
-
-/// Extend common data into PCR
+/// Extend common data into the current and journey PCRs
```
Comment (jhand2) [inline fmc/src/flow/pcr.rs:36]:
```suggestion
/// Extend common data into RT current and journey PCRs
```

File: fmc/src/flow/pcr.rs (hunk 2 above)
Comment (bluegate010) [reply fmc/src/flow/pcr.rs]: Leftover debug lines, removed.

File: fmc/src/flow/pcr.rs (hunk 3 above)
Comment (bluegate010) [reply fmc/src/flow/pcr.rs]: Done.

**Review by jhand2 — APPROVED** [review]

**Review by jhand2 — APPROVED** [review]

**Review by mhatrevi — APPROVED** [review]

**Review by FerralCoder — APPROVED** [review]

**Pull request merged**